CMakeLists.txt

```diff
@@ -26,6 +26,16 @@ include(GNUInstallDirs)
 include(FortranCInterface)
 include(CMakePackageConfigHelpers)
 
+# Requires OpenMP
```

Comment:
@bennybp 
I've previously been using the [standardized autocmake module for omp](https://github.com/psi4/gdma/blob/master/cmake/autocmake_omp.cmake). I tested enabling omp with Intel compilers, and having `ENABLE_OPENMP=ON` automatically switches from `ifcore` to `ifcoremt` just as we want. So since you said `-openmp` triggers `-recursive`, I removed the recursive logic and just put in OMP detection.

@robertodr
You'll want to make analogous changes to [these](https://github.com/psi4/dkh/commit/53c71394d2efb298409c1ca1b9f29fff754e0aae) in PCMSolver to handle threading safely, methinks.